### PR TITLE
Added full adder for more efficient subseed control

### PR DESCRIPTION
**Pull request by paulhandy**

### CODE REVIEW

File: tmath/src/add.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
+use trytes::*;
+use core::cmp::max;
+
+#[inline]
+pub fn add_trits(first: &[Trit], other: &[Trit], out: &mut [Trit]) {
```
Comment (th0br0) [inline tmath/src/add.rs]: Could we instead do this in place? i.e. `pub fn add_trits(lh: &[Trit], rh: &mut [Trit])` and just expect the user to initialise the right hand side?

**Review by th0br0 — CHANGES_REQUESTED** [review]
Looking good but could we move `add_trits` to happen in place (i.e. only take 2 arguments)

**Comment by th0br0** [comment]
oh - and please add some documentation as well.

File: tmath/src/add.rs (hunk 1 above)
Comment (paulhandy) [reply tmath/src/add.rs]: Fair enough to me

**Review by th0br0 — APPROVED** [review]

**Pull request merged**